README.md

```diff
@@ -6,10 +6,83 @@
 
 ## Introduction
 
-Frequenz gRPC API for monitoring and control of microgrids.
+The Frequenz Microgrid API is an API for operating and monitoring electrical
+components and sensors within a microgrid. It offers various functionalities
+related to microgrid management, including metadata retrieval, component
+listing, setting power levels, and controlling the state of components. The API
+is structured around the concept of a microgrid, which is a localized energy
+system composed of various interconnected components and auxiliary sensors.
 
-This repository provides the core gRPC and protobuf spec together with
-support for generating dedicated API packages for various languages.
+The API supports different _categories_ of electrical components and sensors.
+Electrical components include batteries, inverters, meters, and more. Sensors
+include temperature sensors, humidity sensors, and more. Note that each of these
+categories could have their own _types_, e.g., inverters could belong to the
+type "battery" or "PV". This information is stored in the metadata of each
+component and sensor, and is exposed by the API.
+
+The API streams data from the microgrid in real-time, allowing users to monitor
+the status and performance of components and sensors. It also allows users to
+control the microgrid by setting power levels and controlling the state of
+components.
+
+## Objective
+
+The objective of this API is to provide a standardized interface for monitoring
+and controlling components and sensors within a microgrid. The API is designed
+to be vendor-agnostic, allowing users to interact with components from various
+vendors using the same API.
+
+### Key Features
+
+- **Metadata Retrieval:** The API allows users to retrieve metadata about the
+  microgrid, providing information about the microgrid's unique identifier (ID)
+  and geographical coordinates.
+
+- **Component Listing:** Users can list components in the microgrid, optionally
+  filtered by component IDs and categories.
+
+- **Connection Listing:** The API provides a way to list connections between
```

Comment:
I would suggest to introduce the term "component graph" so it connects to the SDK.